apps/speed.c

```diff
@@ -561,52 +639,514 @@ static OPT_PAIR ecdh_choices[] = {
 };
 #endif
 
+#ifndef SIGALRM
+# define COND(d) (count < (d))
+# define COUNT(d) (d)
+#else
+# define COND(c) (run && count<0x7fffffff)
+# define COUNT(d) (count)
+#endif                         /* SIGALRM */
+
+static unsigned char *buf = NULL, *buf2 = NULL;
+static int j;
+static char *engine_id = NULL;
+
+
+#ifndef OPENSSL_NO_MD2
+static unsigned char md2[MD2_DIGEST_LENGTH];
```

Comment:
Why is this defined globally and not local to the EVP_Digest_MD2_loop function? Same comment goes for lots of the declarations for the other "loop" functions below.